### PR TITLE
cmake: Fix benchmarks,test,example compilation options

### PR DESCRIPTION
**Pull request by tmmsartor**
Fix enabling and disabling of BLASFEO utils (benchmarks, test, examples) from CMake build system.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -72,6 +76,13 @@ set(EXT_DEP ON CACHE BOOL "Compile external dependencies in BLASFEO")
 set(MIN_KERNEL_SIZE 2x2 CACHE STRING "Minimum kernel size")
 # set(MIN_KERNEL_SIZE 4x4 CACHE STRING "Minimum kernel size")
 
+# Skip BLASFEO utils for unsupported systems and for Realease build
+if(CMAKE_SYSTEM MATCHES "dSpace" OR CMAKE_BUILD_TYPE MATCHES "Release")
+	set(BLASFEO_EXAMPLES OFF CACHE BOOL "Examples disabled" FORCE)
+	set(BLASFEO_BENCHMARKS OFF CACHE BOOL "Benchmarks disabled" FORCE)
+	set(BLASFEO_EXAMPLES OFF CACHE BOOL "Examples disabled" FORCE)
```
Comment (dkouzoup) [inline CMakeLists.txt]: why BLASFEO_EXAMPLES twice?

**Comment by tmmsartor** [comment]
Sorry. It's a mistake.

**Comment by dkouzoup** [comment]
So does this PR fix my acados shared_libs branch for you locally?

**Comment by dkouzoup** [comment]
Great, thanks a lot! Hopefully @giaf can merge soon.

**Comment by dkouzoup** [comment]
@tmmsartor, travis is still not happy :/

https://travis-ci.org/dkouzoup/acados/jobs/418659195

/usr/bin/x86_64-linux-gnu-ld: CMakeFiles/blasfeo.dir/kernel/avx2/kernel_dgemm_12x4_lib4.S.o: relocation R_X86_64_PC32 against symbol `inner_kernel_dgemm_add_nt_12x4_lib4' can not be used when making a shared object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: final link failed: Bad value
collect2: error: ld returned 1 exit status

**Comment by tmmsartor** [comment]
It seems a different problem :thinking: , I will try BLASFEO compilation on travis as well this time.

**Comment by dkouzoup** [comment]
shall I set it back to generic maybe? I can also try it quickly whenever you have a PR, no problem

**Comment by giaf** [comment]
in gcc you need the -fPIC flag for shared libraries, I don't know how this is done in a portable way in cmake

**Comment by tmmsartor** [comment]
-fPIC was already automatically added by cmake,
but this is another problem, some info [here](https://stackoverflow.com/questions/43367427/32-bit-absolute-addresses-no-longer-allowed-in-x86-64-linux)
It dependents on the version and/or default flags of gcc. 
On my local machine archlinux gcc 8.2 with `--enable-default-pie` is working.
I'm still trying to get the whole picture.

**Comment by mkotlyar** [comment]
@giaf for `-fPIC` in `cmake` you do
```cmake
set(CMAKE_POSITION_INDEPENDENT_CODE ON)
```

**Comment by giaf** [comment]
@tmmsartor do you use this in the cmake?

**Comment by giaf** [comment]
@tmmsartor the link you send is about 32 bit absolute address for static data, like this one
https://github.com/giaf/blasfeo/blob/master/kernel/avx_x86/kernel_dgemm_4x4_lib4.S#L1077
vs relative address rip-relative address, like this one
https://github.com/giaf/blasfeo/blob/master/kernel/avx2/kernel_dgemm_12x4_lib4.S#L5362

For 64-bit code, I already used rip-relative address (unless I forgot somewhere, but by now it should not happen), so this should not be an issue here. I would double-check the compilation with fPIC.

**Comment by tmmsartor** [comment]
@giaf  I tried with `CMAKE_POSITION_INDEPENDENT_CODE` both `ON` and `OFF` and both on my system and on travis this translate to adding the `-fPIC` flag, ([anyway `ON` is the default](https://cmake.org/cmake/help/v3.0/prop_tgt/POSITION_INDEPENDENT_CODE.html)).
On travis system I got the same error in both cases.

**Pull request merged**